virtualmachine/vsphere/vm.go

```diff
@@ -1753,18 +1759,25 @@ func GetVmList(vm *VM, markedTemplate bool) ([]map[string]interface{}, error) {
 			if err != nil {
 				return nil, err
 			}
-			vmMoList[vmMo.Name] = *vmMo
+			vmPropList = append(vmPropList, VmProperties{
+				Name:       vmMo.Name,
+				Properties: *vmMo})
 		}
 	} else {
-		vmMoList, err = getVirtualMachines(vm)
+		if vm.Datacenter == "" {
+			vmPropList, err = getVirtualMachines(vm, true)
```

Comment:
instead of this i would suggest to write it like this : 
allDCs  := false // some better var name
if vm.Datacenter == "" {
 allDCs  := true
} 
vmPropList, err = getVirtualMachines(vm, allDCs)